src/tls_wrap.cc

```diff
@@ -356,6 +357,40 @@ Local<Value> TLSWrap::GetSSLError(int status, int* err, std::string* msg) {
         Local<String> message =
             OneByteString(env()->isolate(), mem->data, mem->length);
         Local<Value> exception = Exception::Error(message);
+        auto obj = exception->ToObject(env()->isolate()->GetCurrentContext())
+          .ToLocalChecked();
+
+        const char* ls = ERR_lib_error_string(e);
+        const char* fs = ERR_func_error_string(e);
+        const char* rs = ERR_reason_error_string(e);
+
+        if (ls != nullptr) obj->Set(env()->isolate()->GetCurrentContext(),
+                 OneByteString(env()->isolate(), "library"),
+                 OneByteString(env()->isolate(), ls)).FromJust();
+        if (fs != nullptr) obj->Set(env()->isolate()->GetCurrentContext(),
+                 OneByteString(env()->isolate(), "function"),
+                 OneByteString(env()->isolate(), fs)).FromJust();
+        if (rs != nullptr) obj->Set(env()->isolate()->GetCurrentContext(),
+                 OneByteString(env()->isolate(), "reason"),
+                 OneByteString(env()->isolate(), rs)).FromJust();
+
+        if (rs != nullptr) {
```

Comment:
Can this be combined into the previous `if (rs != nullptr)` check instead of checking it twice?